pkg/virt-controller/watch/vmi.go

```diff
@@ -634,6 +607,65 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 	return nil
 }
 
+func (c *VMIController) syncReadyConditionFromPod(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) {
+	conditionManager := controller.NewVirtualMachineInstanceConditionManager()
+
+	vmiReadyCond := conditionManager.GetCondition(vmi, virtv1.VirtualMachineInstanceReady)
+	if vmiReadyCond == nil {
+		newCond := virtv1.VirtualMachineInstanceCondition{Type: virtv1.VirtualMachineInstanceReady}
+		vmi.Status.Conditions = append(vmi.Status.Conditions, newCond)
+		vmiReadyCond = &vmi.Status.Conditions[len(vmi.Status.Conditions)-1]
+	}
+
+	setVMICondition := func(status k8sv1.ConditionStatus, reason, message string, lastProbeTime, lastTransitionTime v1.Time) {
+		if vmiReadyCond.Status == status && vmiReadyCond.Reason == reason {
+			return
+		}
+
+		vmiReadyCond.Status = status
+		vmiReadyCond.Reason = reason
+		vmiReadyCond.Message = message
+		vmiReadyCond.LastProbeTime = lastProbeTime
+		vmiReadyCond.LastTransitionTime = lastTransitionTime
+
+		logMessage := "marking VMI as not ready"
+		if status == k8sv1.ConditionTrue {
+			logMessage = "Marking VMI as ready"
+		}
+		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, reason, logMessage)
+		log.Log.Object(vmi).Infof(logMessage + ":" + message)
+	}
+
+	// Keep PodReady condition in sync with the VMI
+	if pod == nil || isTempPod(pod) {
+		setVMICondition(k8sv1.ConditionFalse,
+			virtv1.PodConditionMissingReason,
+			"virt-launcher pod does not exist",
```

Comment:
What would you think about us being a bit more detailed here with the message?

for example, If the VMI is in the early phases, the message should be `virt-launcher pod has not yet been scheduled`, if we're in the later phases of a VMI the message should be something like`virt-launcher pod is terminated`

Reply:
So, we already have several different messages for `Ready: False`:
- "virt-launcher pod does not exist"
- "virt-launcher pod is terminating"
- virt-launcher pod is missing the Ready condition"
- `podReadyCond.Message`, in case virt-launcher pod exists && has this condition set (that's typically ""

I've reworded the first one to "virt-launcher pod has not yet been scheduled" per your suggestion.
Do you see any other case we should have a specific message for?